lib/logstash/filters/fingerprint.rb

```diff
@@ -153,19 +152,22 @@ def fingerprint_ipv4_network(ip_string)
   end
 
   def fingerprint_openssl(data)
+    digest_string = "digest-#{id}"
+    Thread.current[digest_string] ||= select_digest(@method)
```

Comment:
👍

Reply:
I suggest adding a comment about the use of the threadlocal var here, the intent is not self-explanatory; re OpenSSL::Digest::____ objects (jruby openssl) are not thread safe.